synapseutils/sync.py

```diff
@@ -67,15 +67,26 @@ def syncFromSynapse(syn, entity, path=None, ifcollision='overwrite.local', allFi
             print(f.path)
 
     """
+    # initialize the result list
     if allFiles is None: allFiles = list()
+
+    # perform validation check on user input
+    if is_synapse_id(entity):
+        entity = syn.get(entity)
+
```

Comment:
What happens if this is a very large file?  Should we download it then?

Reply:
`syncFromSynapse` does not only work for a container (`project`/ `file`), it also works when users give it a `file` entity. And yes, the expected behavior is to download the file.

The documentation on `entity` parameter does not mention `file`, but the error and the integration test (integration.test_synapseutils.test_syncFromSynapse__given_file_id) indicates that this function should work for files. 

I can change the behavior if you think we should limit this function to only work for container exclusively.